pkg/apis/stork/v1alpha1/action.go

```diff
@@ -34,27 +34,100 @@ type Action struct {
+	FailoverParameter *FailoverParameter `json:"failoverParameter,omitempty"`
+	FailbackParameter *FailbackParameter `json:"failbackParameter,omitempty"`
+}
+
+type FailoverParameter struct {
+	FailoverNamespaces         []string `json:"failoverNamespaces"`
+	MigrationScheduleReference string   `json:"migrationScheduleReference"`
+	DeactivateSource           bool     `json:"deactivateSource"`
+}
+
+type FailbackParameter struct {
+	Namespaces                 []string `json:"namespaces"`
+	MigrationScheduleReference string   `json:"migrationScheduleReference"`
+}
+
 // ActionStatus is the current status of the Action
-type ActionStatus string
+type ActionStatusType string
 
 const (
+	// ActionStatusInitial means Action CR has been created
+	ActionStatusInitial ActionStatusType = ""
 	// ActionStatusScheduled means Action is yet to start
-	ActionStatusScheduled ActionStatus = "Scheduled"
+	ActionStatusScheduled ActionStatusType = "Scheduled"
 	// ActionStatusInProgress means Action is in progress
-	ActionStatusInProgress ActionStatus = "In-Progress"
+	ActionStatusInProgress ActionStatusType = "In-Progress"
 	// ActionStatusFailed means that Action has failed
-	ActionStatusFailed ActionStatus = "Failed"
+	ActionStatusFailed ActionStatusType = "Failed"
 	// ActionStatusSuccessful means Action has completed successfully
-	ActionStatusSuccessful ActionStatus = "Successful"
+	ActionStatusSuccessful ActionStatusType = "Successful"
 )
+
+// ActionStageType is the stage of the action
+type ActionStageType string
+
+const (
+	// ActionStageInitial for when action is created
+	ActionStageInitial ActionStageType = ""
+	// ActionStageScaleDownDestination for scaling down apps in destination
+	ActionStageScaleDownDestination ActionStageType = "ScaleDownDestination"
+	// ActionStageScaleDownSource for scaling down apps in source
+	ActionStageScaleDownSource ActionStageType = "ScaleDownSource"
+	// ActionStageScaleUpDestination for scaling apps in destination
+	ActionStageScaleUpDestination ActionStageType = "ScaleUpDestination"
+	// ActionStageScaleUpSource for scaling apps in source
+	ActionStageScaleUpSource ActionStageType = "ScaleUpSource"
+	// ActionStageFinal is the final stage for action
+	ActionStageFinal ActionStageType = "Final"
+)
+
+// ActionStatus is the status of action operation
+type ActionStatus struct {
+	Stage           ActionStageType  `json:"stage"`
+	Status          ActionStatusType `json:"status"`
+	FinishTimestamp meta.Time        `json:"finishTimestamp"`
+	Summary         *ActionSummary   `json:"summary"`
+	Reason          string           `json:"reason"`
+}
+
+// ActionSummary lists the summary of the action
+type ActionSummary ActionSummaryItem
+
+// ActionSummary has summary for each action type
+type ActionSummaryItem struct {
+	FailoverSummaryItem []*FailoverSummary `json:"failoverSummary,omitempty"`
+	FailbackSummaryItem []*FailbackSummary `json:"failbackSummary,omitempty"`
+}
+
+type FailoverSummary struct {
```

Comment:
json names are incorrect for FailoverSummary and FailbackSummary

Reply:
addressed.